specification/purview/resource-manager/Microsoft.Purview/stable/2021-07-01/purview.json

```diff
@@ -0,0 +1,1958 @@
+        }
+      }
+    },
+    "PrivateLinkResource": {
+      "description": "A privately linkable resource.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "The private link resource identifier.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "The private link resource name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "properties": {
+          "$ref": "#/definitions/PrivateLinkResourceProperties",
+          "description": "The private link resource properties.",
+          "readOnly": true
+        },
+        "type": {
+          "description": "The private link resource type.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "PrivateLinkResourceProperties": {
+      "description": "A privately linkable resource properties.",
+      "type": "object",
+      "properties": {
+        "groupId": {
+          "description": "The private link resource group identifier.",
+          "type": "string",
+          "readOnly": true
+        },
+        "requiredMembers": {
+          "description": "This translates to how many Private IPs should be created for each privately linkable resource.",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "readOnly": true
+        },
+        "requiredZoneNames": {
+          "description": "The required zone names for private link resource.",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "readOnly": true
+        }
+      }
+    },
+    "CheckNameAvailabilityRequest": {
+      "description": "The request payload for CheckNameAvailability API",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Resource name to verify for availability",
+          "type": "string"
+        },
+        "type": {
+          "description": "Fully qualified resource type which includes provider namespace",
+          "type": "string"
+        }
+      }
+    },
+    "CheckNameAvailabilityResult": {
+      "description": "The response payload for CheckNameAvailability API",
+      "type": "object",
+      "properties": {
+        "message": {
+          "description": "Error message",
+          "type": "string"
+        },
+        "nameAvailable": {
+          "description": "Indicates if name is valid and available.",
+          "type": "boolean"
+        },
+        "reason": {
+          "description": "The reason the name is not available.",
+          "enum": [
+            "Invalid",
+            "AlreadyExists"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "reason",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "TrackedResource": {
```

Comment:
Not a must, but consider using resource definition from common-types folder: [https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json)